src/psyclone/psyir/symbols/datatypes.py

```diff
@@ -329,18 +364,19 @@ def __str__(self):
 
         '''
         from psyclone.psyir.symbols import DataSymbol
+        from psyclone.psyir.nodes import DataNode
         dims = []
         for dimension in self.shape:
             if isinstance(dimension, DataSymbol):
                 dims.append(dimension.name)
-            elif isinstance(dimension, int):
+            elif isinstance(dimension, DataNode):
```

Comment:
It's unfortunate that the shapes with Literal strings look much more verbose than the int version we had. But I guess there is still value in having a long, verbose, unambiguous representation. Maybe we should provide both a `_repr_` (long unambiguous) and `_str_` (readable for user/errors, ...) to all PSyIR concepts? (obviously another PR)

Reply:
Yes, I was more aiming at capturing all the information we should in a consistent way. We can display it in a neater more concise form as required and the use of _repr_ and _str_ makes sense to me.